cpp/test/Ice/inactivityTimeout/AllTests.cpp

```diff
@@ -84,19 +84,60 @@ testWithOutstandingRequest(TestIntfPrx p, bool oneway)
     cout << "ok" << endl;
 }
 
+void
+testWithOutstandingOnewaySend(TestIntfPrx p, TestIntfControllerPrx controller)
+{
+    cout << "testing the inactivity timeout with outstanding one-way request to send... "
+         << flush;
+
+    p = p->ice_oneway();
+    p->ice_ping();
+    ConnectionPtr connection1 = p->ice_getCachedConnection();
+
+    // Hold the server adapter and send 10 MB with a oneway request. The sending should block. The connection inactivity
+    // timeout shouldn't be triggered while the connection is waiting for sending the payload.
+    controller->holdAdapter();
+
+    // Send the payload. The payload shouldn't be fully sent until the adapter is resumed.
+    Ice::ByteSeq seq;
+    seq.resize(10 * 1024 * 1024);
+    promise<void> sent;
+    auto future = sent.get_future();
+    p->sendPayloadAsync(seq, []() {}, [](exception_ptr) { test(false); }, [&sent](bool) { sent.set_value(); });
```

Comment:
Since it's a oneway call, it would be simpler to use the future API. Or am I missing something?

Reply:
We wouldn't be able to check that the payload wasn't sent after waiting 100 ms.

Reply:
Ok, you're right.